client/src/js/services/grid/GridGrouping.js

```diff
@@ -98,9 +99,32 @@ function GridGroupingService(GridAggregators, uiGridGroupingConstants, Session,
     }
   }
 
-  function handleBatchSelection (){
+  function handleBatchSelection() {
     var gridApi = this.gridApi;
+    var gridRows = gridApi.selection.getSelectedGridRows();
+    var parents = {};
+
+    var hasSelections = gridApi.selection.getSelectedRows().length > 0;
+
+    gridRows.forEach(function (row) {
+      var parentRow = row.treeNode.parentRow;
+      if (parentRow.treeLevel === 0 && !parents[parentRow.uid]) {
+        parentRow.isSelected = true;
+        parents[parentRow.uid] = parentRow;
+        SELECTED_GROUP_HEADERS = parents;
```

Comment:
Do you need to set this every loop through the `gridRows`?  It looks like what you are really doing is:
```js
gridRows.forEach(function (row) {
  var parentRow = row.treeNode.parentRow;
  if (parentRow.treeLevel === 0 && !parents[parentRow.uid]) {
    parentRow.isSelected = true;
    parents[parentRow.uid] = parentRow;
  }
});

SELECTED_GROUP_HEADERS = parents;
```

If that is true, then it is easier to read (and might be more performant) to remove the `SELECTED_GROUP_HEADERS` variable assignment from the loop.